consensus/consensus.go

```diff
@@ -100,7 +109,19 @@ func NewConsensus(
 	var err error
 	switch cfg.Scheme {
 	case RollDPoSScheme:
-		delegatesByEpochFunc := func(epochNum uint64) ([]string, error) {
+		if ops.bbf == nil {
+			return nil, errors.New("block builder factory is not set")
+		}
+		chanMgr := rolldpos.NewChainManager(bc, sf, ops.bbf)
+		delegatesByEpochFunc := func(epochNum uint64, prevHash []byte) ([]string, error) {
+			fork, serr := chanMgr.Fork(hash.Hash256(prevHash))
+			if serr != nil {
+				return nil, serr
+			}
+			forkSF, serr := fork.StateReader()
+			if err != nil {
```

Comment:
here `err` will always be `nil`, do you mean to check `serr` above?

Reply:
good catch